tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake

```diff
@@ -23,11 +23,11 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-rt711-4ch.bin,DEEPBUFFER_FW_DMA_MS
 "cavs-rt5682\;sof-mtl-max98357a-rt5682\;PLATFORM=mtl,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,\
 PDM1_MIC_B_ENABLE=1,DMIC0_PCM_ID=99,PREPROCESS_PLUGINS=nhlt,\
 NHLT_BIN=nhlt-sof-mtl-max98357a-rt5682.bin,DEEPBUFFER_FW_DMA_MS=10,INCLUDE_ECHO_REF=true,\
-BT_NAME=SSP2-BT,BT_ID=8,BT_PCM_NAME=Bluetooth"
+BT_NAME=SSP2-BT,BT_ID=8,BT_PCM_NAME=Bluetooth,USE_CHAIN_DMA=true"
 
 "cavs-rt5682\;sof-mtl-max98357a-rt5682-ssp2-ssp0\;PLATFORM=mtl,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,\
 PDM1_MIC_B_ENABLE=1,DMIC0_PCM_ID=99,PREPROCESS_PLUGINS=nhlt,\
 NHLT_BIN=nhlt-sof-mtl-max98357a-rt5682.bin,DEEPBUFFER_FW_DMA_MS=10,HEADSET_SSP_DAI_INDEX=2,\
 SPEAKER_SSP_DAI_INDEX=0,HEADSET_CODEC_NAME=SSP2-Codec,SPEAKER_CODEC_NAME=SSP0-Codec,\
-BT_NAME=SSP1-BT,BT_INDEX=1,BT_ID=8,BT_PCM_NAME=Bluetooth,INCLUDE_ECHO_REF=true"
+BT_NAME=SSP1-BT,BT_INDEX=1,BT_ID=8,BT_PCM_NAME=Bluetooth,INCLUDE_ECHO_REF=true,USE_CHAIN_DMA=true"
```

Comment:
is there really a limit to 16 bit only? I would expect 24/32 to be supported as well, no?

Reply:
Ack, will update the commit message.

Reply:
Just did a quick check with aplay, and it seems 24bit is not supported?

```
localhost ~ # aplay -Dhw:0,5 -c8 /home/7-1-24bit.wav -vv
Playing WAVE '/home/7-1-24bit.wav' : Signed 24 bit Little Endian in 3bytes, Rate 48000 Hz, Channels 8
aplay: set_params:1341: Sample format non available
Available formats:
- S16_LE
- S32_LE
```

Reply:
@bardliao is this a real limitation? IIRC everything needs to be converted to 32 bits anyways for SoundWire, no?

Reply:
For S24_LE, It was mentioned here for the limitation:
https://github.com/thesofproject/sof/pull/6524/commits/1edd503a250ac7a0cc9d891624ac3675cfad4b47

Reply:
Ah yes @yongzhi1  24 bit won't work, but S32_LE should work, no?

Reply:
Tried speaker-test, for some reason, the test still used S16_LE even -fS32_LE is provided, also there is no sound (silent playing) coming, investigating ...

```
localhost ~ # speaker-test -Dhw:0,5 -c6 -r48000 -f S32_LE -l1

speaker-test 1.2.1

Playback device is hw:0,5
Stream parameters are 48000Hz, S16_LE, 6 channels
Using 16 octaves of pink noise
Rate set to 48000Hz (requested 48000Hz)
Buffer size range from 32 to 5461
Period size range from 16 to 1365
Using max buffer size 5460
Periods = 4
was set period_size = 1365
was set buffer_size = 5460
 0 - Front Left
 4 - Front Center
 1 - Front Right
 3 - Rear Right
 2 - Rear Left
 5 - LFE
Time per period = 17.798883
```

Reply:
@yongzhi1 it's -FS32_LE

-F,--format	sample format

Classic mistake with speaker-test, I've done this so many times. Really silly...

Reply:
Thanks for catching my mistake, yes, both S16_LE and S32_LE work, I do not have a surround sound system, so only checked 4xch, 6xch and 8xch cases on HDMI TV (with stereo speakers) using sine signal with speaker_test utility.